featurebyte/models/target.py

```diff
@@ -0,0 +1,69 @@
+"""
+This module contains Target related models
+"""
+from __future__ import annotations
+
+from typing import List, Optional
+
+import pymongo
+from pydantic import Field
+
+from featurebyte.models.base import (
+    FeatureByteCatalogBaseDocumentModel,
+    PydanticObjectId,
+    UniqueConstraintResolutionSignature,
+    UniqueValuesConstraint,
+)
+from featurebyte.query_graph.graph import QueryGraph
+
+
+class TargetModel(FeatureByteCatalogBaseDocumentModel):
+    """
+    Model for Target
+
+    id: PydanticObjectId
+        Target id of the object
+    name: str
+        Name of the Target
+    created_at: datetime
+        Datetime when the Target object was first saved or published
+    updated_at: datetime
+        Datetime when the Target object was last updated
+    """
+
+    # Recipe
+    graph: QueryGraph = Field(allow_mutation=False)
+    node_name: str
+
+    description: Optional[str]
```

Comment:
We should remove description here as there will be another task to add description for all existing assets.